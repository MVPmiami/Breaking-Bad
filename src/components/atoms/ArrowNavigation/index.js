import styles from "./style.module.scss";
import backArrow from "./../../../assets/img/leftArr.png";
import rightArrow from "./../../../assets/img/rightArr.png";

const ArrowNavigation = ({
  currentPage,
  paginateToPrevPage,
  paginateToNextPage,
}) => {
  return (
    <div className={styles.arrowNavigation}>
      <img
        src={backArrow}
        className={styles.arrow}
        onClick={() => {
          paginateToPrevPage(currentPage);
        }}
      ></img>
      <img
        src={rightArrow}
        className={styles.arrow}
        onClick={() => {
          paginateToNextPage(currentPage);
        }}
      ></img>
    </div>
  );
};

export default ArrowNavigation;
