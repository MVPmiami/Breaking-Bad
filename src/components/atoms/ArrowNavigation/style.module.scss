.arrowNavigation {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 4rem;
  margin-left: 3rem;
  .arrow {
    &:hover {
      cursor: pointer;
    }
  }
}
