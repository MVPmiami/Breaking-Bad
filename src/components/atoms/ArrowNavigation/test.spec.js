import React from "react";
import ArrowNavigation from "./../../../../src/components/atoms/ArrowNavigation";

const setUp = (props) => shallow(<ArrowNavigation {...props} />);

describe("should render ArrowNavigation component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("should contain .arrowNavigation wrapper", () => {
    const wrapper = component.find(".arrowNavigation");
    expect(wrapper.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it("should call callback of arrowBtnLeft", () => {
    const onBtnClick = jest.fn();
    const wrapper = shallow(
      <ArrowNavigation paginateToPrevPage={onBtnClick} />
    );
    wrapper.find(".arrowLeft").simulate("click");
    expect(onBtnClick).toHaveBeenCalled();
  });

  it("should call callback of arrowBtnRight", () => {
    const onBtnClick = jest.fn();
    const wrapper = shallow(
      <ArrowNavigation paginateToNextPage={onBtnClick} />
    );
    wrapper.find(".arrowRight").simulate("click");
    expect(onBtnClick).toHaveBeenCalled();
  });
});
