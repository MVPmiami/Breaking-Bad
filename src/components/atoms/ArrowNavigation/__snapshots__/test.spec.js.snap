// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ArrowNavigation component should contain .arrowNavigation wrapper 1`] = `
<div
  className="arrowNavigation"
>
  <img
    className="arrow arrowLeft"
    onClick={[Function]}
    src="test-file-stub"
  />
  <img
    className="arrow arrowRight"
    onClick={[Function]}
    src="test-file-stub"
  />
</div>
`;
